### PR TITLE
[KBFS-2830] KBFS Finder Banner in Folder Browser

### PR DESCRIPTION
**Pull request by jzila**
screen shot 2018-03-21 at 4 50 04 pm

screen shot 2018-03-21 at 5 02 03 pm

@keybase/react-hackers @songgao

### CODE REVIEW

File: shared/fs/banner/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  showBanner: boolean,
+  path?: Types.Path,
+  onDismiss?: () => void,
+  onInstall: () => void,
+  onUninstall: () => void,
+  openInFileUI?: () => void,
+}
+
+const Banner = isMobile
+  ? () => <Box />
```
Comment (chrisnojima) [inline shared/fs/banner/index.js]: you can just do () => null

File: shared/fs/banner/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        height: 176,
+        alignItems: 'center',
+        position: 'relative',
+      }
+      let bannerContent
+      if (kbfsEnabled) {
+        bannerContent = (
+          <Box style={globalStyles.flexBoxColumn}>
+            <Text type="Header" style={textStyle}>
+              {onDismiss && 'Yay!'} Keybase is {onDismiss && 'now'} enabled in your {fileUIName}.
```
Comment (chrisnojima) [inline shared/fs/banner/index.js]: you have extra weird spaces if you do it this way (even if some rendering stuff collapses it). Like you want either
```
{'Yay! '}Keybase is {'now '}enabled
```
etc or even better just make one string with templates
```
{`${onDismiss ? 'Yay! ' : ''}Keybase is${onDismiss ? ' now' : ' '}enabled in your ${fileUIName}`}
```

**Comment by jzila** [comment]
@chrisnojima Ready for another look.

File: shared/fs/banner/container.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  ...dispatchProps,
+  openInFileUI: stateProps.kbfsEnabled && path ? () => dispatchProps._openInFileUI(path) : undefined,
+  path,
+})
+
+export default compose(
+  connect(mapStateToProps, mapDispatchToProps, mergeProps),
+  setDisplayName('FilesBanner'),
+  lifecycle({
+    componentDidMount: function() {
```
Comment (chrisnojima) [inline shared/fs/banner/container.js]: i know we have examples of this but you can actually just do ```componentDidMount() {
}```

Comment (jzila) [reply shared/fs/banner/container.js]: In an object?

File: shared/constants/fs.js [hunk 4]
```diff
@@ -75,6 +75,7 @@ export const makeState: I.RecordFactory<Types._State> = I.Record({
   kbfsInstalling: false,
   fuseInstalling: false,
   kextPermissionError: false,
+  showBanner: false,
```
Comment (songgao) [inline shared/constants/fs.js]: It'd be nice if all of FUSE related states can go into a separate `fuse` field which is its own type.

File: shared/fs/banner/container.js (hunk 3 above)
Comment (chrisnojima) [reply shared/fs/banner/container.js]: https://github.com/keybase/client/blob/93298a3c39c40c323634d2dbe33a7ade35611b84/shared/chat/new-team-dialog-container.js#L43

File: shared/fs/banner/index.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        height: 176,
+        alignItems: 'center',
+        position: 'relative',
+      }
+      let bannerContent
+      if (kbfsEnabled) {
+        bannerContent = (
+          <Box style={globalStyles.flexBoxColumn}>
+            <Text type="Header" style={textStyle}>
+              {onDismiss && 'Yay! '}Keybase is {onDismiss && 'now '}enabled in your {fileUIName}.
```
Comment (songgao) [inline shared/fs/banner/index.js]: It seems `onDismiss` is always present. When would it be falsey? Is this for when we have settings that persists the "dismissed" state?

Comment (jzila) [reply shared/fs/banner/index.js]: It's false in `shared/settings/files/index.js`

File: shared/constants/fs.js (hunk 4 above)
Comment (jzila) [reply shared/constants/fs.js]: I'll make a separate refactor PR for that.

**Pull request merged**